### PR TITLE
chore: update Motoko to 0.6.14

### PR DESCRIPTION
**Pull request by ericswanson-dfinity**

### CODE REVIEW

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -6,6 +6,21 @@
...
+Updated Motoko from 0.6.11 to 0.6.14.
+
+* Implement type union/intersection
+* Transform for-loops on arrays into while-loops
+* Tighten typing rules for type annotations in patterns
+* Candid decoding: skip vec any fast
+* Bump up MAX_HP_FOR_GC from 1GB to 3GB
+* Candid decoder: Trap if a principal value is too large
+* Eliminate bignum calls from for-iteration on arrays
+* Make mark-compact the default collector, improve scheduling
```
Comment (crusso) [inline CHANGELOG.adoc:20]:
```suggestion
*  improve scheduling
```

Comment (crusso) [reply CHANGELOG.adoc:20]: That should do - they can look at the Motoko changelog for more detials.

**Review by crusso — APPROVED** [review]

**Review by crusso — APPROVED** [review]
the binaries and moc versions look good to me (motoko-base too).

File: nix/sources.json [hunk 2]
```diff
@@ -83,27 +83,29 @@
         "homepage": null,
         "owner": "dfinity",
         "repo": "motoko-base",
-        "rev": "45bd1abc8e81376d5e1c1dd6322e170e47555c57",
-        "sha256": "18i51cglq8131lyn6w2rjbhx32fbl7y62gga4cs10fdw2fmiw0d8",
+        "rev": "d0a814a3f93a3c168dacbeae2677f2b5d0995c02",
```
Comment (chenyan-dfinity) [inline nix/sources.json:86]: Are we pulling in motoko-base directly now? That means we don't need to package base in motoko release.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -6,6 +6,21 @@
...
+Updated Motoko from 0.6.11 to 0.6.14.
+
+* Implement type union/intersection
+* Transform for-loops on arrays into while-loops
+* Tighten typing rules for type annotations in patterns
+* Candid decoding: skip vec any fast
+* Bump up MAX_HP_FOR_GC from 1GB to 3GB
+* Candid decoder: Trap if a principal value is too large
+* Eliminate bignum calls from for-iteration on arrays
+*  improve scheduling
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:20]:
```suggestion
* Improve scheduling
```

File: CHANGELOG.adoc [hunk 4]
```diff
@@ -6,6 +6,21 @@
...
+* Implement type union/intersection
+* Transform for-loops on arrays into while-loops
+* Tighten typing rules for type annotations in patterns
+* Candid decoding: skip vec any fast
+* Bump up MAX_HP_FOR_GC from 1GB to 3GB
+* Candid decoder: Trap if a principal value is too large
+* Eliminate bignum calls from for-iteration on arrays
+* Improve scheduling
+* Improve performance of bignum equality
+* stable signatures: frontend, metadata, command-line args
```
Comment (ericswanson-dfinity) [inline CHANGELOG.adoc:22]:
```suggestion
* Stable signatures: frontend, metadata, command-line args
```

**Pull request merged**